azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java

```diff
@@ -720,6 +722,16 @@ public VirtualMachineImpl withSpecializedOSDisk(Disk disk, OperatingSystemTypes
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withSpecializedGalleryImageVersion(String galleryImageVersionId) {
+        ImageReference imageReferenceInner = new ImageReference();
+        imageReferenceInner.withId(galleryImageVersionId);
+        this.inner().storageProfile().osDisk().withCreateOption(DiskCreateOptionTypes.FROM_IMAGE);
```

Comment:
This makes `isOSDiskFromImage` true, so no need another bool.

Reply:
Only if it is `specialized`, we need to set OS profile as null. If it is `generalized`, keep as it is.